podcastfy/client.py

```diff
@@ -5,121 +5,173 @@
 from URLs or existing transcript files. It orchestrates the content extraction,
 generation, and text-to-speech conversion processes.
 """
+import copy
 
 import os
 import uuid
 import typer
 import yaml
+
+from podcastfy.aiengines.llm.gemini_langchain import DefaultPodcastifyTranscriptEngine
+from podcastfy.aiengines.tts.base import TTSBackend
+from podcastfy.aiengines.tts.tts_backends import OpenAITTS, ElevenLabsTTS, EdgeTTS
+from podcastfy.core.audio import AudioManager
+from podcastfy.core.character import Character
+from podcastfy.core.content import Content
+from podcastfy.core.podcast import Podcast
+from podcastfy.core.transcript import Transcript
 from podcastfy.content_parser.content_extractor import ContentExtractor
-from podcastfy.content_generator import ContentGenerator
-from podcastfy.text_to_speech import TextToSpeech
+from podcastfy.core.tts_configs import TTSConfig
 from podcastfy.utils.config import Config, load_config
 from podcastfy.utils.config_conversation import (
-    ConversationConfig,
     load_conversation_config,
 )
 from podcastfy.utils.logger import setup_logger
 from typing import List, Optional, Dict, Any
-import copy
-
 
 logger = setup_logger(__name__)
 
 app = typer.Typer()
 
+def create_characters(config: Dict[str, Any]) -> List[Character]:
```

Comment:
Characters are not considering input conversation_config! This is moving us backwards from a functionality perspective.

Reply:
This is just a function to make this character things backward compatible. The Character primitive is very promising for this project IMO.
But I don't think I really understand this comment, why does it remove current functionalities ?